models/marts/ledger_current_state/ttl_current.sql

```diff
@@ -0,0 +1,53 @@
+{{ config(
+    tags = ["current_state"]
+    , materialized='incremental'
+    , unique_key=["key_hash"]
+    , cluster_by= ["key_hash"]
+    )
+}}
+
+/* Finds the latest state of each ttl for contracts or wasms.
+   Ranks each record (grain: one row per contract_id/contract_code_hash)) using
+   the last modified ledger sequence number. */
+
+with
+    current_expiration as (
+        select
+            ttl.key_hash
+            , ttl.live_until_ledger_seq
+            , ttl.last_modified_ledger
+            , ttl.ledger_entry_change
+            , ttl.closed_at
+            , ttl.deleted
+            , ttl.batch_id
+            , ttl.batch_run_date
+            , ttl.batch_insert_ts
+            , row_number()
+                over (
+                    partition by ttl.key_hash
+                    order by ttl.last_modified_ledger desc
```

Comment:
Do we not also need to order by `ledger_entry_change`? If an entry is deleted, won't the last_modified_ledger be identical to the prior record?

Reply:
Ah you're right